patterns/1-initial/source-code-inventory.md

```diff
@@ -0,0 +1,93 @@
+## Title
+
+Enterprise level source code inventory to drive business value and catalyze InnerSource
+
+## Patlet
+
+Enterprises sometimes may manage software, in particular source code as key company asset, with less rigorousness than other IT hardware inventories which may diminish the opportunities to maximize business value and control liability cost. An Enterprise level source code inventory can also exploit opportunities to identify and support potential valuable InnerSource assets.
+
+## Problem
+
+* Some InnerSource stakeholders do not value source code at the same level of other Enterprise assets.
+* Organizations within the Enterprise want to select the right InnerSource projects to maximize outcome.
+* InnerSource Program wants to support the right projects among all the potential candidates.
+* There is no good consolidation of all source code and exploiting business value is ad-hoc.
+
+## Context
+
+* You work on a complex organization under continuous change (e.g., new acquisitions or changing business priorities).
+* You cannot find all source code touched, shared and consumed within your organization.
+* You do not have a clear policy on default sharing level when creating a new project (e.g., Open, Inner or Closed Source).
+* You cannot scan significant parts of your organization's source code looking for duplication, similarity or code smells.
+* You do not know the existing ratios of open source, inner source and closed source and their trend.
+* You cannot measure the diversity of contributions and resulting value for a given project.
+* You cannot identify and optimize tech stack diversity.
+* You cannot identify technical debt and determine the priorities for retirement (e.g., dead APIs/source).
+
+## Forces
+
+* Fragmentation of source code hosting systems in your organization.
+* Ad-hoc source code strategies scattered across your organization.
+* Continuously changing map of the relationships between: projects, repositories, products, tech stacks, domains, solutions, platforms, services, components, sub-systems, people, authors, teams, external repositories.
+* Diverse software culture of teams across your organization (e.g., more open to collaboration or more siloed).
+
+## Solutions
+
+### Set up an Enterprise level live source code inventory
+
+* Combination of manual and automated input data sources to a single source of truth including an API for custom extension
+* Key meta-data about each repository:
+  * Organization
+  * URL
+  * Version control system (e.g., GIT or SVN).
+  * Hosting vendor (e.g., GitHub, Gitlab or BitBucket) and hosting type (e.g., on-prem, private cloud or public cloud).
+  * Sharing level (e.g., Open Source, InnerSource, Closed Source)
+* Visualization in place to list all assets with options to filter based on meta-data
+* Enable access to automated scanning tools
```

Comment:
What do you mean by "scanning tools"?

Reply:
I mean static analysis to help on code quality. I clarify further with examples.